File: sdk/python/examples/quickstart.py
"""Minimal end-to-end example for the Constitutional AIOps Python client.

Run against an instance you can reach. With ``AUTH_REQUIRED`` unset you can omit
the token; otherwise pass one via ``AIOPS_TOKEN``.

    AIOPS_URL=https://your-instance.example.com AIOPS_TOKEN=... python quickstart.py
"""

import os

from constitutional_aiops import AIOpsClient, ConstitutionalRefusal


def main() -> None:
    client = AIOpsClient(
        os.environ.get("AIOPS_URL", "http://localhost:8000"),
        token=os.environ.get("AIOPS_TOKEN"),
    )

    print("Recent incidents:")
    for incident in client.paginate("/incidents/", page_size=25):
        print(f"  {incident['id']:<16} {incident.get('severity','?'):<9} {incident.get('title','')}")

    pending = client.pending_actions()
    print(f"\nPending actions: {pending.get('count', 0)}")

    # A few of the 0.2.0 read helpers.
    print(f"Action stats: {client.action_stats()}")
    print(f"Episodic graph: {client.graph_stats()}")
    print(f"Unread notifications: {client.unread_count().get('count', 0)}")

    # Generative UI (0.3.0): opt-in, cost-fenced explanations of computed data.
    client.set_insight_preferences(enabled=True)
    explained = client.explain("anomaly", {"count": 1, "top": [{"series": "cpu", "z": 3.9}]})
    print("Explain (anomaly):", explained.get("explanation") if explained.get("available") else explained.get("reason"))

    # The MCP tool registry (0.3.0) — the same tools the copilots use.
    print(f"Tools available: {client.tools().get('total', 0)}")

    print("\nChat:")
    try:
        result = client.stream_chat(
            "Summarize the current state of the system.",
            on_delta=lambda token: print(token, end="", flush=True),
        )
        print("\n---")
        print(result.get("message", {}).get("content", ""))
    except ConstitutionalRefusal as refusal:
        print("gate refusal:", refusal.error_code)


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/quickstart.ts
/**
 * Minimal end-to-end example for the Constitutional AIOps TypeScript client.
 * Run against an instance you can reach (Node 18+, global fetch).
 *
 *   AIOPS_URL=https://your-instance.example.com AIOPS_TOKEN=... npx tsx examples/quickstart.ts
 */

import { AIOpsClient, ConstitutionalRefusal } from '../src/index'

async function main(): Promise<void> {
  const client = new AIOpsClient({
    baseUrl: process.env.AIOPS_URL ?? 'http://localhost:8000',
    token: process.env.AIOPS_TOKEN,
  })

  console.log('Recent incidents:')
  for await (const incident of client.paginate('/incidents/', {}, 25)) {
    const inc = incident as { id?: string; severity?: string; title?: string }
    console.log(`  ${inc.id ?? '?'}  ${inc.severity ?? '?'}  ${inc.title ?? ''}`)
  }

  const pending = (await client.pendingActions()) as { count?: number }
  console.log(`\nPending actions: ${pending.count ?? 0}`)

  // A few of the 0.2.0 read helpers.
  console.log('Action stats:', await client.actionStats())
  console.log('Episodic graph:', await client.graphStats())
  const unread = (await client.unreadCount()) as { count?: number }
  console.log(`Unread notifications: ${unread.count ?? 0}`)

  // Generative UI (0.3.0): opt-in, cost-fenced explanations of computed data.
  await client.setInsightPreferences({ enabled: true })
  const explained = (await client.explain('anomaly', { count: 1, top: [{ series: 'cpu', z: 3.9 }] })) as {
    available?: boolean
    explanation?: string
    reason?: string
  }
  console.log('Explain (anomaly):', explained.available ? explained.explanation : explained.reason)

  // The MCP tool registry (0.3.0) — the same tools the copilots use.
  const toolList = (await client.tools()) as { total?: number }
  console.log(`Tools available: ${toolList.total ?? 0}`)

  console.log('\nChat:')
  try {
    const result = await client.streamChat('Summarize the current state of the system.', {
      onDelta: (token) => process.stdout.write(token),
    })
    const message = result.message as { content?: string } | undefined
    console.log('\n---\n' + (message?.content ?? ''))
  } catch (error) {
    if (error instanceof ConstitutionalRefusal) console.log('gate refusal:', error.errorCode)
    else throw error
  }
}

void main()
